### DISCUSSION TITLE
Connecting a form input to monboDB query

### DISCUSSION THREAD

**Discussion by benfranklin23** [Budibase/budibase#10945] [Help]
Hello, I can't seem to understand how to add custom data (say input from a form) into a mongoDB query. Looking at the docs "https://docs.budibase.com/docs/mongodb" it seems like adding bindings is how you do this. However in the docs it says "Now, wherever you use the created query as a data source in a [Data Provider](https://docs.budibase.com/docs/data-provider), you can configure the bindings directly from there, using the gearwheel behind the provider selection." When I try to add a data provider, I do not see my query in the list of options. Am I looking in the wrong place or adding the wrong component? Again, my goal here is to take form input and turn it into a post request into my mongodb database, (username, password for example to add a "user" object to a collection).

**Comment by melohagan** [comment]
Hey @benfranklin23 

Sorry for the late response! Whenever you configure bindings with the cog icon in a Data provider, that is specifically for **Read** queries.

In your case you want to POST (write), therefore this page should be more useful: https://docs.budibase.com/docs/data-sources#tutorial-posting-form-data-to-a-custom-query

The example given is SQL, but it's the same principle for your MongoDB queries.

There was also a video released recently which should help! https://youtu.be/OjGZ841KgBk
